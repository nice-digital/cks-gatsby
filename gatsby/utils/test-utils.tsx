import React, { ReactElement } from "react";
import { render, RenderResult } from "@testing-library/react";
import {
	createMemorySource,
	createHistory,
	LocationProvider,
	History,
} from "@reach/router";

// Util for rendering a component wrapped within Reach Router's LocationProvider
// See https://testing-library.com/docs/example-reach-router
export const renderWithRouter = (
	ui: ReactElement,
	{ route = "/", history = createHistory(createMemorySource(route)) } = {}
): { history: History } & RenderResult => ({
	...render(<LocationProvider history={history}>{ui}</LocationProvider>),
	// adding `history` to the returned utilities to allow us
	// to reference it in our tests (just try to avoid using
	// this to test implementation details).
	history,
});
