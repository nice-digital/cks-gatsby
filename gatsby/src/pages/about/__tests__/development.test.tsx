/* eslint-disable testing-library/no-node-access */
import React from "react";
import { textContentMatcher } from "test-utils";
import { render, waitFor, screen } from "@testing-library/react";

import DevelopmentPage from "../development";

describe("Development page", () => {
	beforeEach(() => {
		render(<DevelopmentPage />);
	});

	describe("SEO", () => {
		it("should set page title", async () => {
			await waitFor(() => {
				expect(document.title).toContain("Development process | About CKS");
			});
		});

		it("should set meta description", async () => {
			await waitFor(() => {
				expect(
					document.querySelector("meta[name='description']")
				).toHaveAttribute(
					"content",
					"The CKS development process, including new topics, topic updates and the CKS process guide"
				);
			});
		});
	});

	describe("Breadcrumbs", () => {
		it.each([
			["NICE", "https://www.nice.org.uk/"],
			["CKS", "/"],
			["About CKS", "/about/"],
		])("Breadcrumbs (%s)", (breadcrumbText, expectedHref) => {
			expect(
				screen.queryByText(textContentMatcher(breadcrumbText), {
					selector: ".breadcrumbs a",
				})
			).toHaveAttribute("href", expectedHref);
		});

		it("should render current page breadcrumb without link", () => {
			expect(
				screen.getByText(textContentMatcher("Development process"), {
					selector: ".breadcrumbs span",
				})
			).toBeTruthy();
		});
	});

	describe("Page header", () => {
		it("should render heading 1 with Development process", () => {
			expect(
				screen.queryByText("Development process", {
					selector: "h1",
				})?.tagName
			).toBe("H1");
		});
	});

	describe("Body", () => {
		it("should match snapshot for 'Help develop CKS' panel", () => {
			expect(
				screen.getByText("Help develop CKS").parentElement
			).toMatchSnapshot();
		});

		it("should match snapshot for 'CKS process guide' panel", () => {
			expect(
				screen.getByText("CKS process guide").parentElement
			).toMatchSnapshot();
		});

		it("should match snapshot for main body content", () => {
			expect(screen.getByText("New topics").parentElement).toMatchSnapshot();
		});
	});
});
