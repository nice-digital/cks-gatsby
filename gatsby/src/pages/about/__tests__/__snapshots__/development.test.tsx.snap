// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Development page Body should match snapshot for 'CKS process guide' panel 1`] = `
<div
  class="panel"
>
  <h2
    class="h3"
  >
    CKS process guide
  </h2>
  <p>
    Detailed information on the CKS development process:
  </p>
  <p>
    <a
      class="btn btn--secondary"
      download=""
      href="https://www.nice.org.uk/media/default/About/what-we-do/Evidence%20Services/CKS/CKS-guide.pdf"
    >
      CKS process guide (PDF, 400 KB)
    </a>
  </p>
</div>
`;

exports[`Development page Body should match snapshot for 'Help develop CKS' panel 1`] = `
<div
  class="panel"
>
  <h2
    class="h3"
  >
    Help develop CKS
  </h2>
  <p>
    CKS topics rely on feedback from healthcare professionals, patients groups, patients and carers to ensure they are accurate and relevant.
  </p>
  <ul
    aria-label="How to help develop CKS"
  >
    <li>
      <a
        href="https://cks.clarity.co.uk/get-involved/"
      >
        Help develop CKS topics
      </a>
    </li>
    <li>
      <a
        href="https://www.nice.org.uk/cks-feedback"
      >
        Send feedback on CKS topics
      </a>
    </li>
  </ul>
</div>
`;

exports[`Development page Body should match snapshot for main body content 1`] = `
<div
  class="body"
  data-g="12 md:7 lg:8"
>
  <h2>
    New topics
  </h2>
  <p>
    The topic development panel take into account the following when developing new topics:
  </p>
  <ul
    aria-label="What's taken into account when developing new topics"
  >
    <li>
      enquiries sent to NICE CKS
    </li>
    <li>
      GP curriculum
    </li>
    <li>
      new NICE clinical guidelines
    </li>
    <li>
      new pressures affecting primary care as highlighted by the Department of Health and Public Health England.
    </li>
  </ul>
  <p>
    Up to 10 new CKS topics are produced each year.
  </p>
  <h2>
    Updates
  </h2>
  <p>
    Topics are fully reviewed and updated every 5 years. We will update a topic sooner if significant new evidence emerges such as:
  </p>
  <ul
    aria-label="Reasons for updating a topic sooner"
  >
    <li>
      publication of new/updated NICE guidance
    </li>
    <li>
      new updated national polices or safety information
    </li>
    <li>
      changes to products licenses and device availability
    </li>
    <li>
      other important new evidence.
    </li>
  </ul>
  <p>
    65+ CKS updates are produced each year.
  </p>
  <h2>
    Topic development
  </h2>
  <p>
    CKS topics are developed for NICE by Clarity Informatics Ltd using
     
    <a
      href="https://www.nice.org.uk/about/what-we-do/accreditation"
    >
      an accredited development process
    </a>
    .
  </p>
  <p>
    Topics are developed and reviewed by a multidisciplinary team consisting of:
  </p>
  <ul
    aria-label="Roles that develop and review topics"
  >
    <li>
      clinicians
    </li>
    <li>
      pharmacists
    </li>
    <li>
      pharmacologists
    </li>
    <li>
      information specialists
    </li>
    <li>
      informatics analysts
    </li>
    <li>
      technical authors.
    </li>
  </ul>
  <p>
    The process used to develop and review CKS topics includes the following:
  </p>
  <ul
    aria-label="Stages to develop and review CKS topics"
  >
    <li>
      topic development and selection
    </li>
    <li>
      scoping
    </li>
    <li>
      developing clinical questions and scenarios
    </li>
    <li>
      literature searching
    </li>
    <li>
      evaluating and summarising the evidence
    </li>
    <li>
      developing clinical recommendations
    </li>
    <li>
      external consultation
    </li>
    <li>
      release.
    </li>
  </ul>
  <h3>
    Evidence-based
  </h3>
  <p>
    CKS topics are developed and updated using the best available evidence. High-quality secondary evidence from NICE accredited resources (for example, NICE guidance and Cochrane systematic reviews) is identified first, and primary research and expert opinion sought where necessary.
  </p>
  <h3>
    Relevant information
  </h3>
  <p>
    The development process has a number of checks to make sure that the information is relevant and appropriate to primary care professionals:
  </p>
  <ul
    aria-label="CKS process checks for relevant information"
  >
    <li>
      thorough scoping and review process
    </li>
    <li>
      input from healthcare professionals with primary care experience
    </li>
    <li>
      consultation with GPs, pharmacists, lay members
    </li>
    <li>
      feedback received from external reviews.
    </li>
  </ul>
  <p>
    More detailed information on the way CKS topics are developed can be found in the
     
    <a
      href="https://www.nice.org.uk/media/default/About/what-we-do/Evidence%20Services/CKS/CKS-guide.pdf"
    >
      CKS process guide (PDF, 400 KB)
    </a>
    .
  </p>
  <p>
    <a
      class="btn btn--secondary"
      download=""
      href="https://cks.clarity.co.uk/get-involved/"
    >
      Help develop CKS topics
    </a>
  </p>
</div>
`;
